third_party/terraform/utils/config.go.erb

```diff
@@ -668,7 +668,11 @@ func (c *Config) getTokenSource(clientScopes []string) (oauth2.TokenSource, erro
 
 	log.Printf("[INFO] Authenticating using DefaultClient...")
 	log.Printf("[INFO]   -- Scopes: %s", clientScopes)
-	return googleoauth.DefaultTokenSource(context.Background(), clientScopes...)
+	if c, err := googleoauth.DefaultTokenSource(context.Background(), clientScopes...); err == nil {
+		return c, err
+	} else {
+		return c, fmt.Errorf("Attempted to load application default credentials since neither `credentials` nor `access_token` was set in the provider block.  No credentials loaded.  Consider running 'gcloud auth application-default login.  Original error: %w", err)
```

Comment:
Would this work?
"To use your existing credentials, run 'gcloud auth application-default login'."
Also, I think you or your IDE is using two spaces after a period. Not sure if this is intentional.

Reply:
Yep, that's me!  In all my typed communication my entire life to date.  :)  It's what I learned in school, seems a lot of work to change a lifelong habit.  Causing an issue?

For the comment, I don't know that it's your existing credentials, maybe "your user account's credentials"?

Reply:
Oh wild, Github just knocks that down to one space, I have never noticed that before.  How odd!

Reply:
Huh, just looked it up and it seems like some people *really* hate it.  Never knew it was such a contentious issue.  I guess I can configure vim to correct me when I do it in a user-visible string, if need be - I imagine you're finding it distracting?

Reply:
I think it's important to call out that these are *gcloud* credentials and not just misc existing credentials. People get really confused when they think they're using credentials that are specific to them and they get an error that they have to enable an API in a project they don't own.

Reply:
@danawillow _gcloud_ credentials make sense, LGTM :)
@ndmckinley Not distracting, just something I haven't seen much so I wasn't sure if it was intentional.